client/landing/gutenboarding/gutenboard.tsx

```diff
@@ -56,25 +50,30 @@ export function Gutenboard() {
 	const [ isEditorSidebarOpened, updateIsEditorSidebarOpened ] = useState( false );
 	const toggleGeneralSidebar = () => updateIsEditorSidebarOpened( isOpen => ! isOpen );
 
-	const onboardingState = useSelect( select => select( STORE_KEY ).getState() );
+	const { siteVertical } = useSelect( select => select( STORE_KEY ).getState() );
 
-	// FIXME: Quick'n'dirty step state, replace with router
-	const [ currentStep, setStep ] = useState( Steps.IntentGathering );
-	const goToNextStep = stepCompleted[ currentStep ]( onboardingState )
-		? () => setStep( step => step + 1 )
-		: undefined;
-	const goToPrevStep = currentStep > 0 ? () => setStep( step => step - 1 ) : undefined;
+	// @TODO: This is currently needed in addition to the routing (inside the Onboarding Block)
+	// for the 'Back' and 'Next' buttons in the header. If we remove those (and move navigation
+	// entirely into the block), we'll be able to remove this code.
+	const r = useRouteMatch( '*' );
+	let next: undefined | string;
+	let prev: undefined | string;
+	switch ( r?.url ) {
+		case '/':
+			if ( siteVertical ) {
+				next = '/design';
+			}
+			break;
 
-	const onboardingBlock = useRef( createBlock( name, { step: currentStep } ) );
+		case '/design':
+			prev = '/';
+			break;
+	}
 
-	const { updateBlockAttributes } = useDispatch( 'core/block-editor' );
-	useEffect(
-		() =>
-			void updateBlockAttributes( onboardingBlock.current.clientId, {
-				step: currentStep,
-			} ),
-		[ currentStep, updateBlockAttributes ]
-	);
+	// We're caching the block via `useRef` in order to prevent re-renders
+	// which would collide with the routing done inside of the block
+	// (and would lead to weird mounting/unmounting behavior).
```

Comment:
"caching" sounds like an optimization. This is a necessary bit that keeps a reference to the block so it is persistent across renders.

Reply:
"persisting"? :grimacing: